src/codeu/chat/server/JSON.java

```diff
@@ -68,141 +69,182 @@ public Model readFromFile (String file) throws IOException
             // will be used and returned.
             System.out.println("Invalid file");
         }
-       if (jp!=null) {
+        if (jp!=null) {
             // If the file exists, then the file will be parsed
-           // in this set of code
-           jp.nextToken();
+            // in this set of code
+            jp.nextToken();
             while (jp.nextToken() != JsonToken.END_OBJECT)
             // This ensures that we haven't reached the end of the
             // file because the data is being parsed in such a way
             // that this END_OBJECT specifically relates to the
             // very last } in the json file
             {
-            if(jp.getText().equals("users")) {
-                // If the object type represents a User, then
-                // specific steps will take place to create
-                // a User variable
-                jp.nextToken();
-                while (jp.nextToken() != JsonToken.END_ARRAY) {
+                if(jp.getText().equals("users")) {
+                    // If the object type represents a User, then
+                    // specific steps will take place to create
+                    // a User variable
+                    //jp.nextToken();
+                    Time time = null;
                     String name = "";
                     Uuid id = null;
-                    Time time = null;
-                    jp.nextToken();
-                    if (jp.getText().equals("name")) {
-                        jp.nextToken();
-                        name = jp.getText();
-                    }
-                    jp.nextToken();
-                    if (jp.getText().equals("uuid")) {
-                        jp.nextToken();
-                        try {
-                            id = Uuid.parse(jp.getText());
-                        } catch (IOException e) {
-                            System.out.println("Invalid Uuid");
+                    while (jp.nextToken() != JsonToken.END_ARRAY) {
+                        if (jp.getText().equals("name")) {
+                            jp.nextToken();
+                            name = jp.getText();
+                            //jp.nextToken();
+                        }
+                        if (jp.getText().equals("uuid")) {
```

Comment:
Doesn't this mean that the order still matters? It needs to be `name`, then `uuid`, then `creationTime`, etc... right? Maybe I'm just missing something? (I haven't really worked with jackson before, so that's pretty likely!)